### PR TITLE
fix bce_with_logits with legacy reduce

### PR DESCRIPTION
**Pull request by ailzhang**
Fix #12624 . internal usecase of legacy `reduce`. 
Add test in test_nn

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -6928,6 +6928,20 @@ def bceloss_weights_no_reduce_scalar_test():
     )
 
 
+def bce_with_logistic_legacy_enum_test():
```
Comment (SsnL) [inline test/test_nn.py]: a lot of these code is really legacy!

**Review by ssnl — APPROVED** [review]
Let's get this in! :)

But we should also add legacy enum tests to all our loss functions.
